mantidproject/mantid
# DataProcessorUI: make post-processing step optional

First steps towards making post-processing optional in the `DataProcessorUI`.

No post-processing means no groups, which means that some of the existing table editing features (such as `PlotGroup`, `GroupRows`, `ExpandSelection`, etc) should not be available. For this reason, the presenter now has to tell the view which actions should be shown in the toolbar (therefore all actions have been removed from the .ui file) and menus. 

In addition, no post-processing means that the tree is a one-level tree (as opposed to the current two-level tree where parent items are groups and child items are runs within a group) where parent items are runs with no children. This means that the presenter has to be able to deal with two different types of trees, a two-level tree when we have groups and a one-level tree when we do not have groups.

For the reasons above, I created a new abstract base class, `TreeManager`, that is intended to be implemented by two derived classes, `OneLevelTreeManager` and `TwoLevelTreeManager`, that will deal with different tree models and implement actions in different ways.

This is work in progress. I just wanted to share what I've done to make sure it is the right approach.

**To test:**

Read the interface [documentation](http://docs.mantidproject.org/nightly/interfaces/ISIS_Reflectometry.html) and test as in https://github.com/mantidproject/mantid/pull/16941. Make sure there are no changes in functionality.

Fixes #16632.

_Does not need to be in the release notes._ This is a purely internal change with no effect on functionality.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
